packages/@n8n/benchmark/scripts/n8nSetups/scaling/docker-compose.yml

```diff
@@ -0,0 +1,176 @@
+services:
+  redis:
+    image: redis:6-alpine
+    restart: always
+    ports:
+      - 6379:6379
+    healthcheck:
+      test: ['CMD', 'redis-cli', 'ping']
+      interval: 1s
+      timeout: 3s
+
+  postgres:
+    image: postgres:16
+    user: ${RUN_USER_AND_GROUP}
+    restart: always
+    environment:
+      - POSTGRES_DB=n8n
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=password
+      - PGDATA=/var/lib/postgresql/data/pgdata
+    volumes:
+      - ${RUN_DIR}/postgres:/var/lib/postgresql/data
+    healthcheck:
+      test: ['CMD-SHELL', 'pg_isready -U postgres']
+      interval: 5s
+      timeout: 5s
+      retries: 10
+
+  n8n_worker1:
+    image: ghcr.io/n8n-io/n8n:${N8N_VERSION:-latest}
+    user: ${RUN_USER_AND_GROUP}
+    environment:
+      - N8N_DIAGNOSTICS_ENABLED=false
+      - N8N_USER_FOLDER=/n8n/worker1
+      - N8N_ENCRYPTION_KEY=very-secret-encryption-key
```

Comment:
AFAIK the only way to define variables in a Docker compose file is via environment variables. For simplicity I feel it's nicer to define the value inline instead of adding indirection that passes it from outside.

Reply:
Extracted out from the compose file